microshift.sh

```diff
@@ -0,0 +1,120 @@
+  if ! sudo subscription-manager status >& /dev/null ; then
+     sudo subscription-manager register
+  fi
+else
+  echo "This script only works for RHEL-8"
+fi
+
+run_preflight_checks ${BUNDLE_TYPE}
+rm -fr ${INSTALL_DIR} && mkdir ${INSTALL_DIR}
+
+sudo virsh destroy ${CRC_VM_NAME} || true
+sudo virsh undefine --nvram ${CRC_VM_NAME} || true
+sudo rm -fr /var/lib/libvirt/images/${CRC_VM_NAME}.qcow2
+sudo rm -fr /var/lib/libvirt/images/microshift-installer-*.iso
+
+# Download the oc binary for specific OS environment
+OC=./openshift-clients/linux/oc
+download_oc
+
+# Generate a new ssh keypair for this cluster
+# Create a 521bit ECDSA Key
+rm id_ecdsa_crc* || true
+ssh-keygen -t ecdsa -b 521 -N "" -f id_ecdsa_crc -C "core"
+
+function configure_host {
+    sudo dnf install -y git osbuild-composer composer-cli ostree rpm-ostree \
+      cockpit-composer cockpit-machines bash-completion podman genisoimage \
+      createrepo yum-utils selinux-policy-devel jq wget lorax
+    sudo systemctl enable osbuild-composer.socket --now
+    sudo systemctl enable cockpit.socket --now
+    sudo firewall-cmd --add-service=cockpit --permanent
+    sudo chmod 0755 $HOME
+}
+
+function enable_repo {
+    # Add respective repo to fetch the microshift and dependency packages
+    sudo subscription-manager repos \
+       --enable rhocp-4.12-for-rhel-8-$(uname -i)-rpms \
+       --enable fast-datapath-for-rhel-8-$(uname -i)-rpms
+}
+
+function download_microshift_rpm {
+    local pkgDir=$(mktemp -p /tmp -d tmp-rpmXXX)
+    sudo dnf download --downloadonly --downloaddir ${pkgDir} microshift microshift-networking microshift-selinux >/dev/null 2>&1
+    sudo chmod 0777 ${pkgDir}
+    echo ${pkgDir}
+}
+
+function create_iso {
+    local pkgDir=$1
+    rm -fr microshift
+    # Clone microshift repo
+    git clone -b release-4.12 https://github.com/openshift/microshift.git
+    cp podman_changes.ks microshift/
+    pushd microshift
+    sudo chmod 0755 $HOME
+    ssh-keygen -t ecdsa -b 521 -N "" -f id_ecdsa_crc -C "core"
+    sed -i '/# customizations/,$d' scripts/image-builder/config/blueprint_v0.0.1.toml
+    sed -i 's/redhat/core/g' scripts/image-builder/config/kickstart.ks.template
+    sed -i '/--bootproto=dhcp/a\network  --hostname=api.crc.testing' scripts/image-builder/config/kickstart.ks.template
+    sed -i '$e cat podman_changes.ks' scripts/image-builder/config/kickstart.ks.template
+    scripts/image-builder/cleanup.sh -full
+    scripts/image-builder/build.sh -microshift_rpms ${pkgDir} -pull_secret_file ${OPENSHIFT_PULL_SECRET_PATH} -authorized_keys_file $(realpath ../id_ecdsa_crc.pub)
+    popd
+    rm -fr $pkgDir
+}
+
+enable_repo
+configure_host
+microshift_pkg_dir=$(download_microshift_rpm)
+create_iso ${microshift_pkg_dir}
+sudo cp microshift/_output/image-builder/microshift-installer-*.iso /var/lib/libvirt/images/microshift-installer.iso
+
+OPENSHIFT_RELEASE_VERSION=$(sudo dnf list | awk "/^microshift\./ {print \$2;}")
```

Comment:
Since the rpms are already downloaded, you can directly use `rpm -qp --qf='%{VERSION}' microshift...rpm`

Reply:
it is installed, don't know the download location for image-builder scripts also `dnf` step was mentioned in the document before ( it is removed due to removal of offine container image section)

Reply:
`download_microshift_rpm` does `sudo dnf download --downloadonly --downloaddir ${pkgDir} microshift microshift-release-info microshift-networking microshift-selinux >/dev/null 2>&1`, `rpm -qp` will tell rpm to look at a locally downloaded `.rpm` file, so it should be possible to use `rpm` directly rather than slowly going through `dnf`. And we'd be sure the version matches what we are going to install.